server/valgrind-out.txt

```diff
@@ -0,0 +1,231 @@
+--159719-- REDIR: 0x4917f80 (libc.so.6:rawmemchr) redirected to 0x483f220 (_vgnU_ifunc_wrapper)
+--159719-- REDIR: 0x4a0a810 (libc.so.6:__strrchr_avx2) redirected to 0x484e810 (rindex)
+--159719-- REDIR: 0x4a0a600 (libc.so.6:__strchrnul_avx2) redirected to 0x4853330 (strchrnul)
+--159719-- REDIR: 0x49120a0 (libc.so.6:malloc) redirected to 0x4848820 (malloc)
+--159719-- REDIR: 0x4a0a9e0 (libc.so.6:__strlen_avx2) redirected to 0x484ed10 (strlen)
+--159719-- REDIR: 0x4a0d980 (libc.so.6:__mempcpy_avx_unaligned_erms) redirected to 0x4853440 (mempcpy)
+--159719-- REDIR: 0x4a0e180 (libc.so.6:__memset_avx2_unaligned_erms) redirected to 0x4852770 (memset)
+--159719-- REDIR: 0x49123e0 (libc.so.6:free) redirected to 0x484b210 (free)
+--159719-- REDIR: 0x4915fc0 (libc.so.6:__GI_strstr) redirected to 0x4853ae0 (__strstr_sse2)
+--159719-- REDIR: 0x4a064c0 (libc.so.6:__memchr_avx2) redirected to 0x484fd50 (memchr)
+--159719-- REDIR: 0x4a0d9c0 (libc.so.6:__memcpy_avx_unaligned_erms) redirected to 0x4852880 (memmove)
+--159719-- REDIR: 0x4a0a380 (libc.so.6:__strchr_avx2) redirected to 0x484e9f0 (index)
+--159719-- REDIR: 0x4912740 (libc.so.6:realloc) redirected to 0x484dc50 (realloc)
+--159719-- REDIR: 0x4a06c80 (libc.so.6:__memcmp_avx2_movbe) redirected to 0x4852010 (bcmp)
+--159719-- REDIR: 0x4a0beb0 (libc.so.6:__strcpy_avx2) redirected to 0x484ee60 (strcpy)
+--159719-- REDIR: 0x4913520 (libc.so.6:calloc) redirected to 0x484d9d0 (calloc)
+--159719-- REDIR: 0x4a05b40 (libc.so.6:__strcmp_avx2) redirected to 0x484fbd0 (strcmp)
+==159745== 
+==159745== HEAP SUMMARY:
+==159745==     in use at exit: 20,480 bytes in 1 blocks
+==159745==   total heap usage: 23 allocs, 22 frees, 50,502 bytes allocated
+==159745== 
+==159745== Searching for pointers to 1 not-freed blocks
+==159745== Checked 107,280 bytes
+==159745== 
+==159745== 20,480 bytes in 1 blocks are still reachable in loss record 1 of 1
+==159745==    at 0x4848899: malloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
+==159745==    by 0x109BFE: main (aesdsocket.c:241)
+==159745== 
+==159745== LEAK SUMMARY:
+==159745==    definitely lost: 0 bytes in 0 blocks
+==159745==    indirectly lost: 0 bytes in 0 blocks
+==159745==      possibly lost: 0 bytes in 0 blocks
+==159745==    still reachable: 20,480 bytes in 1 blocks
+==159745==         suppressed: 0 bytes in 0 blocks
+==159745== 
+==159745== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
+==159761== 
+==159761== HEAP SUMMARY:
+==159761==     in use at exit: 20,480 bytes in 1 blocks
+==159761==   total heap usage: 30 allocs, 29 frees, 88,522 bytes allocated
+==159761== 
+==159761== Searching for pointers to 1 not-freed blocks
+==159761== Checked 107,280 bytes
+==159761== 
+==159761== 20,480 bytes in 1 blocks are still reachable in loss record 1 of 1
+==159761==    at 0x4848899: malloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
+==159761==    by 0x109BFE: main (aesdsocket.c:241)
```

Comment:
This is probably highlighting the fact that you don't have cleanup handling after handling exit signals.